### PR TITLE
Check dependies in functions

### PR DESCRIPTION
**Pull request by mumichae**
Check if optional dependencies are available in their respective functions and control behaviour when import fails.

Checks for:

- [x] all integration methods
- [x] kBET
- [x] LISI

### CODE REVIEW

File: tests/metrics/test_kbet.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+def test_kbet(adata_pca):
+    score = scIB.me.kBET(
+        adata_pca,
+        batch_key='batch',
+        label_key='celltype',
+        embed='X_pca'
+    )
+    score = 1 - np.nanmean(score['kBET'])
+    LOGGER.info(f"score: {score}")
+    assert np.isnan(score)
```
Comment (LuckyMD) [inline tests/metrics/test_kbet.py:15]: I think `np.nanmean` ensures this is never nan. Apparently a runtime warning is output if `np.nanmean` gets a vector of all `NaN`s.

**Review by LuckyMD — COMMENTED** [review]
Really nice overall, but the kBET test might need some work. Have you run pipeline test on this?

File: tests/metrics/test_kbet.py (hunk 1 above)
Comment (mbuttner) [reply tests/metrics/test_kbet.py:15]: For small cell types, where kBET might not be applicable, a `NaN` value is returned. `np.mean` fails when it encounters `NaN`, `np.nanmean` ignores them.

Comment (LuckyMD) [reply tests/metrics/test_kbet.py:15]: This test prob also needs to work when kBET can't be run on any celltype... This might be the case, no?

Comment (mbuttner) [reply tests/metrics/test_kbet.py:15]: Indeed. In the case of all `NaN`, `np.nanmean` returns only `nan`.

Comment (LuckyMD) [reply tests/metrics/test_kbet.py:15]: Ah... wait... a RunTimeWarning doesn't mean an error. All good here.

File: tests/metrics/test_kbet.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def test_kbet(adata_pca):
+    score = scIB.me.kBET(
+        adata_pca,
+        batch_key='batch',
+        label_key='celltype',
+        embed='X_pca'
+    )
+    score = 1 - np.nanmean(score['kBET'])
```
[the remark this answers is not in the document]
Comment (mumichae) [reply tests/metrics/test_kbet.py:13]: yes, would be moved to the kbet fix

File: scIB/metrics/lisi.py [hunk 3]
```diff
@@ -326,7 +326,12 @@ def lisi_graph_py(
     # create evenly split chunks if n_obs is divisible by n_chunks (doesn't really make sense on 2nd thought)
     n_splits = n_chunks - 1
     args_int = [cpp_file_path.as_posix(), mtx_file_path, dir_path, str(n_neighbors), str(n_splits), str(subset)]
-    subprocess.run(args_int)
+    try:
+        subprocess.run(args_int)
+    except Exception as e:
```
[the remark this answers is not in the document]
Comment (mumichae) [reply scIB/metrics/lisi.py:331]: I didn't know exactly which exception would be caused on a different architecture so I went with a general one for now

**Comment by mumichae** [comment]
The test is tailored to the  Github actions workflow. In order for kBET to work properly, we'd need a proper R installation in the CI, which would be something for a separate workflow and PR.

**Review by LuckyMD — APPROVED** [review]

File: tests/metrics/test_kbet.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuckyMD) [reply tests/metrics/test_kbet.py:13]: sounds good to me!

**Comment by LuckyMD** [comment]
Thanks @mumichae !

**Pull request merged**